geopandas/vectorized.pyx

```diff
@@ -1093,6 +1103,23 @@ cdef geom_type(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
     return out
 
 
+cdef cy_unary_union(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
+    cdef GEOSContextHandle_t handle
+    cdef GEOSGeometry *collection
+    cdef GEOSGeometry *out
+    cdef size_t n = geoms.size
+
+    handle = get_geos_context_handle()
+
+    with nogil:
+        collection = GEOSGeom_createCollection_r(handle, GEOS_MULTIPOLYGON,
+                                                 <GEOSGeometry **> geoms.data,
+                                                 n)
+        out = GEOSUnaryUnion_r(handle, collection)
+
+    return geom_factory(<np.uintp_t> out)
```

Comment:
They also don't seem to delete the collection:

https://github.com/Toblerity/Shapely/blob/6baa7a566e69847f0cddfe173b72e303d08e8cac/shapely/ops.py#L149-L150